src/netex_idf/modes.rs

```diff
@@ -12,109 +12,282 @@
     pub commercial_mode: (&'static str, &'static str),
 }
 
+impl NTFSMode {
+    pub fn new(
+        physical_mode: (&'static str, &'static str),
+        commercial_mode: (&'static str, &'static str),
+    ) -> Self {
+        NTFSMode {
+            physical_mode,
+            commercial_mode,
+        }
+    }
+}
+
 lazy_static! {
-    pub static ref MODES: HashMap<&'static str, ModeNetexIDF> = {
+    pub static ref MODES: HashMap<IDFMMode, NTFSMode> = {
         let mut m = HashMap::new();
         m.insert(
-            "air",
-            ModeNetexIDF {
-                physical_mode: ("Air", "Avion"),
-                commercial_mode: ("Air", "Avion"),
-            },
+            IDFMMode::Air,
+            NTFSMode::new(("Air", "Avion"), ("Air", "Avion")),
+        );
+        m.insert(IDFMMode::Bus, NTFSMode::new(("Bus", "Bus"), ("Bus", "Bus")));
+        m.insert(
+            IDFMMode::Coach,
+            NTFSMode::new(("Coach", "Autocar"), ("Coach", "Autocar")),
         );
         m.insert(
-            "bus",
-            ModeNetexIDF {
-                physical_mode: ("Bus", "Bus"),
-                commercial_mode: ("Bus", "Bus"),
-            },
+            IDFMMode::Ferry,
+            NTFSMode::new(("Ferry", "Ferry"), ("Ferry", "Ferry")),
         );
         m.insert(
-            "coach",
-            ModeNetexIDF {
-                physical_mode: ("Coach", "Autocar"),
-                commercial_mode: ("Coach", "Autocar"),
-            },
+            IDFMMode::Metro,
+            NTFSMode::new(("Metro", "Métro"), ("Metro", "Métro")),
         );
         m.insert(
-            "ferry",
-            ModeNetexIDF {
-                physical_mode: ("Ferry", "Ferry"),
-                commercial_mode: ("Ferry", "Ferry"),
-            },
+            IDFMMode::RailShuttle,
+            NTFSMode::new(
+                ("RailShuttle", "Orlyval, CDG VAL"),
+                ("RailShuttle", "Orlyval, CDG VAL"),
+            ),
         );
         m.insert(
-            "metro",
-            ModeNetexIDF {
-                physical_mode: ("Metro", "Métro"),
-                commercial_mode: ("Metro", "Métro"),
-            },
+            IDFMMode::RailSuburban,
+            NTFSMode::new(
+                ("LocalTrain", "Train Transilien"),
+                ("LocalTrain", "Train Transilien"),
+            ),
         );
         m.insert(
-            "rail",
-            ModeNetexIDF {
-                physical_mode: ("LocalTrain", "Train régional / TER"),
-                commercial_mode: ("LocalTrain", "Train régional / TER"),
-            },
+            IDFMMode::RailRegional,
+            NTFSMode::new(("Train", "TER / Intercités"), ("regionalRail", "TER")),
         );
         m.insert(
-            "trolleyBus",
-            ModeNetexIDF {
-                physical_mode: ("Tramway", "Tramway"),
-                commercial_mode: ("TrolleyBus", "TrolleyBus"),
-            },
+            IDFMMode::RailShuttle,
```

Comment:
```suggestion
            IDFMMode::RailInterregional,
```

Reply:
Thanks. Done in 262c93cc6a7f0d2c514fd2ed1ac466210b57b96c.